aea/cli/common.py

```diff
@@ -83,19 +80,33 @@ class Context(object):
     def __init__(self, cwd: str = "."):
         """Init the context."""
         self.config = dict()  # type: Dict
-        self.agent_loader = ConfigLoader("aea-config_schema.json", AgentConfig)
-        self.skill_loader = ConfigLoader("skill-config_schema.json", SkillConfig)
-        self.connection_loader = ConfigLoader(
-            "connection-config_schema.json", ConnectionConfig
-        )
-        self.contract_loader = ConfigLoader(
-            "contract-config_schema.json", ContractConfig
-        )
-        self.protocol_loader = ConfigLoader(
-            "protocol-config_schema.json", ProtocolConfig
-        )
         self.cwd = cwd
 
+    @property
+    def agent_loader(self) -> ConfigLoader:
```

Comment:
we avoid multiple instantiations of the configuration loaders by having them in a single class `ConfigLoaders` (see below).